Fix incorrect python script directory in run_inference.sh

dluks opened the pull request:
When running with Docker, the input was being set to `$SOURCE_DIR` (the input data directory) (see #6). This fails as the input directory is actually bound to:
```console
/home/nibio/mutable-outside-world/bucket_in_folder
```
while the scripts live in:
```console
/home/nibio/mutable-outside-world
```

This fixes this by setting a new `SCRIPT_DIR` variable which points to the actual scripts directory and sets the `PYTHONPATH` accordingly.

julesmorel reviewed (changes requested):
Suggested Change on Line 56:

The current line:
`cp "$SOURCE_DIR/conf/eval.yaml" "$DEST_DIR"`

I propose updating this to:
`cp "$SCRIPT_DIR/conf/eval.yaml" "$DEST_DIR"`

dluks commented:
It seems that the configuration isn't quite set up to be particularly boilerplate-friendly for generalized use cases, so I've added a small change to the default `eval.yaml` config and `.hydra/overrides.yaml`.

I've also updated `modify_eval.py` to raise a `SystemExit` when no `.ply` files are found in the output directory.
